src/rgw/rgw_lc.cc

```diff
@@ -828,15 +824,47 @@ int RGWLC::handle_multipart_expiration(rgw::sal::Bucket* target,
   params.ns = RGW_OBJ_NS_MULTIPART;
   params.access_list_filter = &mp_filter;
 
+  const auto event_type = rgw::notify::ObjectExpirationAbortMPU;
+  std::string version_id;
+
   auto pf = [&](RGWLC::LCWorker* wk, WorkQ* wq, WorkItem& wi) {
+    int ret{0};
     auto wt = boost::get<std::tuple<lc_op, rgw_bucket_dir_entry>>(wi);
     auto& [rule, obj] = wt;
     if (obj_has_expired(this, cct, obj.meta.mtime, rule.mp_expiration)) {
       rgw_obj_key key(obj.key);
       std::unique_ptr<rgw::sal::MultipartUpload> mpu = target->get_multipart_upload(key.name);
-      int ret = mpu->abort(this, cct, null_yield);
+      std::unique_ptr<rgw::sal::Object> sal_obj
+	= target->get_object(key);
+      std::unique_ptr<rgw::sal::Notification> notify
+	= driver->get_notification(
+	  this, sal_obj.get(), nullptr, event_type,
+	  target, lc_id,
+	  const_cast<std::string&>(target->get_tenant()),
+	  lc_req_id, null_yield);
+
+      ret = notify->publish_reserve(this, nullptr);
+      if (ret < 0) {
+				ldpp_dout(wk->get_lc(), 0)
+					<< "ERROR: reserving persistent notification for abort_multipart_upload, ret=" << ret
+					<< ", thread:" << wq->thr_name()
+					<< ", meta:" << obj.key
+					<< dendl;
+      }
+
+      ret = mpu->abort(this, cct, null_yield);
```

Comment:
i understand that the operation should succeed even if the notification failed?

Reply:
yes, I think so

Reply:
well, it looks like I had the opposite intuition in the original PR adding expire notificaitons.  It's definitely not acceptable to be inconsistent.  I think this really should be a policy decision--I can see a reasonable site policy preferring to reduce data by default, and also one that would prefer notifications to be reliable at the risk of data growth

Reply:
agree we must be consistent here.
as for the policy, if the users does not need elivery guarantee for the notification. they can configure `ack-level=none` for the topic, this would mean that the queue entry is cleared once the notification is sent (not waiting for an ack), which means that the queue does get full even if the broker is down. in such a case `publish_reserve()` would fail only if there is a RADOS issue.

Reply:
does rgw similarly refuse to perform an i/o request altogether if publish_reserve() fails?

Reply:
yes, when `publish_reserve()` fails with queue full indication, we reply "BUSY" to the client

Reply:
ok